packages/optimizely-sdk/lib/plugins/odp_manager/index.browser.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2023, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import packageJSON from '../../../package.json';
+
+import BrowserAsyncStorageCache from '../key_value_cache/browserAsyncStorageCache';
+
+import { BrowserRequestHandler } from './../../utils/http_request_handler/browser_request_handler';
+import { LRUCache } from '../../utils/lru_cache';
+
+import { VuidManager } from './../vuid_manager/index';
+
+import { OdpConfig } from './../../core/odp/odp_config';
+import { OdpManager } from '../../core/odp/odp_manager';
+import { OdpEvent } from '../../core/odp/odp_event';
+import { invalidOdpDataFound } from '../../core/odp/odp_utils';
+import { OdpEventApiManager } from '../../core/odp/odp_event_api_manager';
+import { OdpEventManager, STATE } from '../../core/odp/odp_event_manager';
+import { OdpSegmentApiManager } from '../../core/odp/odp_segment_api_manager';
+import { OdpSegmentManager } from '../../core/odp/odp_segment_manager';
+import { getLogger, LogHandler, LogLevel } from '../../modules/logging';
+import { ERROR_MESSAGES, JAVASCRIPT_CLIENT_ENGINE, LOG_MESSAGES, ODP_USER_KEY } from '../../utils/enums';
+import PersistentKeyValueCache from '../key_value_cache/persistentKeyValueCache';
+
+interface BrowserOdpManagerConfig {
+  disable: boolean;
+  logger?: LogHandler;
+  odpConfig?: OdpConfig;
+  segmentsCache?: LRUCache<string, string[]>;
+  eventManager?: OdpEventManager;
+  segmentManager?: OdpSegmentManager;
+}
+
+// Client-side Browser Plugin for ODP Manager
+export class BrowserOdpManager extends OdpManager {
+  static cache = new BrowserAsyncStorageCache();
+  vuid?: string;
+
+  constructor({ disable, logger, odpConfig, segmentsCache, eventManager, segmentManager }: BrowserOdpManagerConfig) {
+    const browserLogger = logger || getLogger();
+
+    const browserRequestHandler = new BrowserRequestHandler(browserLogger);
+    const browserClientEngine = JAVASCRIPT_CLIENT_ENGINE;
+    const browserClientVersion = packageJSON.version;
+
+    super({
+      disable,
+      requestHandler: browserRequestHandler,
+      logger: browserLogger,
+      clientEngine: browserClientEngine,
+      clientVersion: browserClientVersion,
+      odpConfig,
+      segmentsCache,
```

Comment:
```suggestion
      segmentsCache || BrowerLRUCache(),
```
Or not sure if you have a plan to inject BrowserLRUCache() always.